test/fixtures/hdfs-fixture/src/main/java/org/elasticsearch/test/fixtures/hdfs/HdfsFixture.java

```diff
@@ -0,0 +1,444 @@
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.CommonConfigurationKeysPublic;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.permission.AclEntry;
+import org.apache.hadoop.fs.permission.AclEntryType;
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.ha.BadFencingConfigurationException;
+import org.apache.hadoop.ha.HAServiceProtocol;
+import org.apache.hadoop.ha.HAServiceTarget;
+import org.apache.hadoop.ha.NodeFencer;
+import org.apache.hadoop.ha.ZKFCProtocol;
+import org.apache.hadoop.ha.protocolPB.HAServiceProtocolClientSideTranslatorPB;
+import org.apache.hadoop.hdfs.DFSConfigKeys;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+import org.apache.hadoop.hdfs.MiniDFSNNTopology;
+import org.apache.hadoop.hdfs.server.namenode.ha.HATestUtil;
+import org.apache.hadoop.hdfs.tools.DFSHAAdmin;
+import org.apache.hadoop.security.SecurityUtil;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.junit.Assume;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class HdfsFixture extends ExternalResource {
+
+    private TemporaryFolder temporaryFolder = new TemporaryFolder();
+    private MiniDFSCluster dfs;
+    private String haNameService;
+    private Supplier<String> principalConfig = null;
+    private Supplier<Path> krb5Conf = null;
+    private Supplier<Path> keytab = null;
+    private Configuration cfg;
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private Configuration haConfiguration;
+    private int explicitPort = findAvailablePort();
+
+    public HdfsFixture withHAService(String haNameService) {
+        this.haNameService = haNameService;
+        return this;
+    }
+
+    public HdfsFixture withKerberos(Supplier<String> principalConfig, Supplier<Path> krb5confFile, Supplier<Path> keytabFile) {
+        this.principalConfig = principalConfig;
+        this.krb5Conf = krb5confFile;
+        this.keytab = keytabFile;
+        return this;
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        temporaryFolder.create();
+        // TODO fix assumeHdfsAvailable();
+        // assumeHdfsAvailable();
+        // UserGroupInformation.reset();
+        startMinHdfs();
+        super.before();
```

Comment:
This is unncessary since the superclass impl is a noop.

Reply:
removed